examples/fit_ODNP_data.py

```diff
@@ -0,0 +1,135 @@
+    R1p = find_file(data_info['filename'], exp_type=data_info["data_dir"], expno=f"{data_info['nodename']}/R1p")
+except:
+    T1p = find_file(data_info['filename'], exp_type=data_info['data_dir'], expno=f"{data_info['nodename']}/T1p")
+    R1p = T1p**-1
+# }}}
+# {{{Find the index where the return powers begin
+flip_idx = np.where(np.diff(Ep.fromaxis("power").data) < 0)[0][0] + 1
+# }}}
+with figlist_var() as fl:
+    # {{{Plot Ep
+    fl.next("Integrated Enhancement")
+    fl.plot(
+        Ep["power", :flip_idx],
+        "o",
+        label="Progressive Saturation Data",
+        capsize=6,
+        alpha=0.5,
+    )
+    fl.plot(
+        Ep["power", flip_idx:],
+        "rx",
+        label="Returning Power Check",
+        capsize=6,
+        alpha=0.5,
+    )
+    # }}}
+    # {{{ plot R1p data
+    fl.next(r"$R_{1}(p)$")
+    fl.plot(R1p, "o", label="Experimental Data")
+    # }}}
+    # {{{ fit krho inverse with two degrees of freedom and then apply to fit R1p
+    # {{{load in T100 dataset
+    T10_p = loadmat(
+        search_filename(
+            T100_info['filename'], exp_type=T100_info['data_dir'], unique=True
+        )
+    )["a"][0, :]
+    R10_p = R1p.fromaxis("power").eval_poly(T10_p, "power") ** -1
+    # }}}
+    krho_inv = Ep.get_prop("acq_params")["concentration"] / (R1p - R10_p)
+    krho_inv_coeff = krho_inv.polyfit("power", order=1)
+    krho_inv_fine = R1p.fromaxis("power").eval_poly(krho_inv_coeff, "power")
+    M0, A, phalf, p = symbols("M0 A phalf power", real=True)
+    R1p_expression = (T10_p[0] + T10_p[1] * p) ** -1 + (
+        Ep.get_prop("acq_params")["concentration"]
+        / (krho_inv_coeff[0] + krho_inv_coeff[1] * p)
+    )
+    R1p_fit = lambdify(p,R1p_expression)
+    fl.plot(R1p_fit(R1p.fromaxis('power')), ls=":", color="k", label="Fit", alpha=0.5)
+    plt.ylabel(r"$R_{1} / s^{-1}$")
+    plt.xlabel("Power / W")
+    # }}}
+    # {{{ Fit E(p)
+    fl.next("Integrated Enhancement")
+    sp = p / (p + phalf)
+    Ep_fit = lmfitdata(Ep["power", :flip_idx])
+    # Symbolic expression for Ep that is used in the symbolic function for the fitting of E(p)
+    Ep_fit.functional_form = M0 - ((M0 * A * sp) / R1p_expression)
+    #generate a guess for the A parameter of the fit based on the normalized enhancement weighted by the relaxation rate. The bounds for the fit are then set to center around this value.
+    A_guess = (
+        1
+        - (
+            R1p["power", 0].data * (Ep["power", flip_idx].data / Ep["power", 0].data)
+        ).real
+    )
+    Ep_fit.set_guess(
+        M0=dict(value=Ep["power", 0].real.item(), min=1e4, max=11e4),
+        A=dict(value=A_guess, min=0.5 * A_guess, max=3 * A_guess),
+        phalf=dict(value=0.2, min=0.1, max=0.4),
+    )
+    Ep_fit.settoguess()
+    Ep_fit.fit()
+    thisfit = Ep_fit.eval(100)
+    fl.plot(thisfit, ls=":", color="k", label="Fit", alpha=0.5)
+    ksig = (
+        Ep_fit.output("A") * Ep.get_prop("acq_params")["guessed_MHz_to_GHz"] * 1e-3
```

Comment:
Why are you using the guess of the PPT value here?

Reply:
there is no "actual ppt" value in the config file. the guessed is used to set the bounds of the field sweep and then we plug in the actual value we find before we run ODNP - but I can edit the append_conc... to be edit_h5_acq_params and make that an option?